File: php/features.php
<?php
declare(strict_types=1);

// TemporaryEmail SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TemporaryEmailFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TemporaryEmailBaseFeature();
            case "test":
                return new TemporaryEmailTestFeature();
            default:
                return new TemporaryEmailBaseFeature();
        }
    }
}

File: py/temporaryemail_sdk/features.py
# TemporaryEmail SDK feature factory

from temporaryemail_sdk.feature.base_feature import TemporaryEmailBaseFeature
from temporaryemail_sdk.feature.test_feature import TemporaryEmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TemporaryEmailBaseFeature(),
        "test": lambda: TemporaryEmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
